### PR TITLE
Added queryable extensions methods for data

### LINKED ISSUE #3396: Extension Methods for Data on IQueryable
**Problem**
To take advantage of Filters, Projects, and Sorting, the Resolver has to return IQueryable. Often I need to post-process, map or enrich the result from the database. It could be from an external service call or because I need to validate some data object ownership.

**Solution**
Add a set of HC Extension that takes an IQueryable and returns an IQueryable like the well-known linq extensions. It could look like this:

```c#
dbContext.Person
  .Filter(resolverContext)  // HC version of Where
  .Projection(resolverContext)  // HC version of Select including subtree
  .Sort(resolverContext)  // HC version of OrderBy
  .SingleOrDefaultAsync()
```

It could also be nice to do the same for MongoDB. The Filter extension would return IFindFluent

### PR DESCRIPTION
**Pull request by PascalSenn**
Closes #3396

### CODE REVIEW

File: src/HotChocolate/Data/src/Data/Projections/Expressions/Extensions/QueryableProjectionExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Collections.Generic;
+using HotChocolate.Resolvers;
+
+namespace HotChocolate.Data.Projections.Expressions
+{
+    public static class QueryableProjectExtensions
```
Comment (michaelstaib) [inline src/HotChocolate/Data/src/Data/Projections/Expressions/Extensions/QueryableProjectionExtensions.cs:7]: XML Documentation?

File: src/HotChocolate/Data/src/Data/Projections/Expressions/QueryableProjectionProvider.cs [hunk 2]
```diff
@@ -1,59 +1,96 @@
...
 using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
 using System.Linq;
 using System.Linq.Expressions;
 using System.Threading.Tasks;
 using HotChocolate.Resolvers;
 
 namespace HotChocolate.Data.Projections.Expressions
 {
+    [return: NotNullIfNotNull("input")]
```
Comment (michaelstaib) [inline src/HotChocolate/Data/src/Data/Projections/Expressions/QueryableProjectionProvider.cs:11]: Does this work?

File: src/HotChocolate/Data/src/Data/Sorting/Expressions/Extensions/QueryableSortExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Collections.Generic;
+using HotChocolate.Resolvers;
+
+namespace HotChocolate.Data.Sorting.Expressions
+{
+    public static class QueryableSortExtensions
```
Comment (michaelstaib) [inline src/HotChocolate/Data/src/Data/Sorting/Expressions/Extensions/QueryableSortExtensions.cs:7]: XML Docs

File: src/HotChocolate/Data/test/Data.Filters.Tests/Attributes/CatchErrorMiddlewareAttribute.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
+using System;
+using System.Reflection;
+using HotChocolate.Types;
+using HotChocolate.Types.Descriptors;
+
+namespace HotChocolate.Data.Filtering
+{
+    public class CatchErrorMiddlewareAttribute : ObjectFieldDescriptorAttribute
```
Comment (michaelstaib) [inline src/HotChocolate/Data/test/Data.Filters.Tests/Attributes/CatchErrorMiddlewareAttribute.cs:8]: XML Documentation

File: src/HotChocolate/Data/src/Data/Projections/Expressions/QueryableProjectionProvider.cs (hunk 2 above)
Comment (PascalSenn) [reply src/HotChocolate/Data/src/Data/Projections/Expressions/QueryableProjectionProvider.cs:11]: Nope

**Review by michaelstaib — APPROVED** [review]

**Pull request merged**